flutter/packages/duck_router/lib/src/configuration.dart

```diff
@@ -5,7 +5,7 @@ import 'package:duck_router/src/interceptor.dart';
 import 'package:duck_router/src/location.dart';
 
 /// Handler for when the app gets a deeplink.
-typedef DuckRouterDeepLinkHandler = List<Location> Function(
+typedef DuckRouterDeepLinkHandler = List<Location>? Function(
```

Comment:
todo: documentation needs updating I think, now that this interface is getting more complicated. Should mention the options the user has on how to handle.

Reply:
Added information about the different possibilities here